test/test_expanded_weights.py

```diff
@@ -0,0 +1,121 @@
+
+from functools import partial
+import torch
+import torch.nn as nn
+from torch.testing._internal.common_device_type import instantiate_device_type_tests
+from torch.testing._internal.common_utils import TestCase, freeze_rng_state, make_tensor, run_tests
+from torch._expanded_weights import ExpandedWeight
+from torch.nn.utils._stateless import per_sample_call
+from torch.testing._internal.common_nn import TestBase, module_tests, new_module_tests
+
+from torch.overrides import handle_torch_function, has_torch_function_variadic
+
```

Comment:
Some more suggestions for tests:
- create an ExpandedWeight object and check that all of its fields (shape, dtype, etc) work.
- Test error cases: If the user does something weird with an ExpandedWeight we should try to error out gracefully. E.g. if the user tries to do torch.add(regular_tensor, ExpandedWeight) then that is probably a problem because we can't infer where the batch dim on regular_tensor is.

Reply:
Added (1) as `TestExpandedWeightAttributes` and switched the fallback case to be a testing error case